File: benchmarks/reverse/reverse.py
import sys
import time

N = int(sys.argv[1])

t0 = time.perf_counter()

arr = []

i = 0
while i < N:
  arr.append(i)
  i += 1

i= 0
while i < (N / 2):
  temp = arr[i]
  arr[i] = arr[N - i - 1]
  arr[N - i - 1] = temp 
  i += 1

t1 = time.perf_counter()
print("Execution took (ms) " + str((t1 - t0) * 1000))

File: benchmarks/reverse/reverse.lua

local t0 = os.clock()
local N = tonumber(arg[1])

arr = {}

local i = 0
while i < N do
    table.insert(arr, i)
    i = i + 1
end

i = 0
while i < (N / 2) do
    temp = arr[i + 1]
    arr[i + 1] = arr[N - i]
    arr[N - i] = temp
    i = i + 1
end

local t1 = os.clock()

print("Execution took (ms) " .. ((t1 - t0) * 1000))